x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/TrainedModelConfig.java

```diff
@@ -243,7 +243,8 @@ public boolean isAvailableWithLicense(XPackLicenseState licenseState) {
         }
 
         // The model license does not matter, this is the highest licensed level
-        if (licenseState.isActive() && XPackLicenseState.isPlatinumOrTrialOperationMode(licenseState.getOperationMode())) {
+        if (licenseState.isActive() && XPackLicenseState.isAllowedByOperationMode(
+            licenseState.getOperationMode(), License.OperationMode.PLATINUM, true)) {
```

Comment:
Strictly speaking, this check is incorrect because there is a race condition between the 2 calls. I don't think we should be perpetuating this style of call, and should actually push it to a single call that passes in the `licenseState` object and checks both `active` and operation mode.